in_toto/model.go

```diff
@@ -31,6 +35,24 @@ type Key struct {
 	Scheme              string   `json:"scheme"`
 }
 
+func validateKeyId(keyId string) error {
+	keyIdFormatCheck, _ := regexp.MatchString("^[a-fA-F0-9]+$", keyId)
+	if !keyIdFormatCheck {
+		return fmt.Errorf("'Key ID' has invalid format")
+	}
+	return nil
+}
+
+func validatePubKey(key Key) error {
+	if key.KeyVal.Private != "" {
+		return fmt.Errorf("private key found")
+	}
+	if key.KeyVal.Public == "" {
+		return fmt.Errorf("public key cannot be empty")
+	}
+	return nil
+}
+
 /*
 Signature represents a generic in-toto signature that contains the identifier
 of the Key, which was used to create the signature and the signature data.  The
```

Comment:
What about `validateSignature`? See `securesystemslib.formats.SIGNATURE_SCHEMA`.

Reply:
It's now being validated as part of `validateMetablock`, should that be broken off into `validateSignature` and then called there? Are there other places where `validateSignature` can come in handy?

Reply:
Yes, please make it a separate function, even if it's only called inside `validateMetablock` for now.

We still need to decide which parts of an in-toto implementation should be part of a public API (see https://github.com/in-toto/in-toto/issues/277), but IMO it's likely that there will be API functions that take `Signature` structs as argument, and we should always validate all arguments in public facing functions.